api/user.go

```diff
@@ -2538,3 +2540,27 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, GetProtocol(r)+"://"+r.Host, http.StatusFound)
 	}
 }
+
+func userTyping(req *model.WebSocketRequest, responseData map[string]interface{}) *model.AppError {
+	var ok bool
+	var teamId string
+	if teamId, ok = req.Data["team_id"].(string); !ok || len(teamId) != 26 {
+		return NewInvalidWebSocketParamError(req.Action, "team_id")
+	}
+
+	var channelId string
+	if channelId, ok = req.Data["channel_id"].(string); !ok || len(channelId) != 26 {
+		return NewInvalidWebSocketParamError(req.Action, "channel_id")
+	}
+
+	var parentId string
+	if parentId, ok = req.Data["parent_id"].(string); !ok {
+		parentId = ""
+	}
+
+	event := model.NewWebSocketEvent(teamId, channelId, req.Session.UserId, model.WEBSOCKET_EVENT_TYPING)
```

Comment:
I think as @hmhealey says it doesn't work cross-team DMs because the event is for a specific `teamId`, If I understand correctly if we remove `teamId` it should send the message for every team right?